org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/ProceedOnErrorTransTypes.java

```diff
@@ -41,5 +42,12 @@ public void visitApply(JCMethodInvocation tree) {
 		}
 		super.visitApply(tree);
 	}
+	
+	@Override
+	public void visitIdent(JCIdent tree) {
+		if (tree.type.getEnclosingType() == null) {
+			return;
+		}
```

Comment:
missing call to `super.visitIdent(tree)` in the regular case.

Reply:
:facepalm: